library/ssl_tls13_generic.c

```diff
@@ -2926,51 +2916,53 @@ int mbedtls_ssl_read_certificate_process( mbedtls_ssl_context* ssl )
 #if defined(MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED)
     if( ret == SSL_CERTIFICATE_EXPECTED )
     {
-        /* Reading step */
-        if( ssl->keep_current_message == 0 )
-        {
-            if( ( ret = mbedtls_ssl_read_record( ssl, 0 ) ) != 0 )
-            {
-                /* mbedtls_ssl_read_record may have sent an alert already. We
-                   let it decide whether to alert. */
-                MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_read_record", ret );
-                goto cleanup;
-            }
-        }
-        else
-        {
```

Comment:
I'm not able to find corresponding `keep_current_message` logic in the new code(non MPS part). `keep_current_message` is useful when the server doesn't send certificate request and the certificate msg(not certificate request) is consumed by `ssl_certificate_request_coordinate`. So we use this flag to indicate that we don't read it again.

Reply:
This code is functionally equivalent to a single `mbedtls_ssl_read_record()`, which now happens in the `fetch()` sub-routine.